### PR TITLE
Symbolic: Bug fix in div/mod for negative min/max

### PR DESCRIPTION
**Pull request by ronak69**
This fixes Issue #1316.

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/shape/symbolic.py                 2               2
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

File: tinygrad/shape/symbolic.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def __mod__(self, b:int):
     assert b > 0
     if b == 1: return NumNode(0)
     if self.min >= 0 and self.max < b: return self
-    if self.min < 0: return (self - ((self.min//b)*b)) % b
+    if self.min < 0: return create_node(ModNode(create_rednode(SumNode, [self, NumNode(-1*(self.min//b)*b)]), b))
```
Comment (geohot) [inline tinygrad/shape/symbolic.py:80]: I really doubt this should be calling create node

Comment (ronak69) [reply tinygrad/shape/symbolic.py]: Did you mean create "red" node? I fixed that. (Nested SumNodes resulted if `self` was a SumNode)

Removing calls to `create_node` will be ok if NumNode implements it's own `__mod__` method.

**Comment by geohot** [comment]
Closing, it's still calling `create_node`

**Comment by ronak69** [comment]
@geohot

> Closing, it's still calling `create_node`

Why do you "doubt" calling `create_node`?

Just `assert b > 0 and self.min >= 0` if you really don't want to support negatives instead of having buggy logic.

**Comment by ronak69** [comment]
I asked you for your reason, i give mine:

I tried to fix an error silently. Bad.

The issue with line 80:
* In both div and mod, an offset (`-((self.min//b)*b)`) is added to make the numerator positive
* Final expression is `(self - ((self.min//b)*b)) % b`
* But that invokes `SumNode`'s `__mod__` method, which will distribute the mod into sum and you end up with `self % b` which is what you started with so: infinite recursion.

Solutions:
* div solves it by using the `factoring_allowed` function parameter of `SumNode`'s div. But it's mod doesn't have it so maybe add it? (big diff)
* Just create `ModNode` explicitly. Which is what i did.

I wrapped `create_node` around it mainly because that is what the next line after it does. But functionally it makes `NumNode`s if min/max are equal.

**Pull request closed without merging**